packages/fx-core/src/component/generator/generator.ts

```diff
@@ -44,17 +45,18 @@ export class Generator {
     actionContext?: ActionContext
   ): Promise<Result<undefined, FxError>> {
     const appName = ctx.projectSetting?.appName;
-    const projectId = ctx.projectSetting?.projectId;
-    const nameReplaceMap = { ...{ appName: appName }, ...ctx.templateVariables };
-    const dataReplaceMap = { ...{ projectId: projectId }, ...nameReplaceMap };
+    const replaceMap = {
+      ...{ appName: appName, SafeProjectName: convertToAlphanumericOnly(appName) },
```

Comment:
Maybe also add ProjectName as it is used for filename replacement.
And I wonder if we can just remove appName from the map since there isn't such a placeholder in v3 templates.

Reply:
https://github.com/OfficeDev/TeamsFx/blob/58a540929b3707b28925270131cae4dd2c436615/templates/scenarios/command-and-response-ts/package.json.tpl#L2 Will this be removed in the future?

Reply:
I think so. Let me follow up this.